mkosi/__init__.py

```diff
@@ -1787,7 +1780,9 @@ def finalize_default_initrd(
         "--acl", str(config.acl),
         *(f"--package={package}" for package in config.initrd_packages),
         *(f"--volatile-package={package}" for package in config.initrd_volatile_packages),
-        *(["--package-directory", str(package_dir)] if package_dir else []),
+        *(f"--package-directory={d}" for d in config.package_directories),
+        *(f"--volatile-package-directory={d}" for d in config.volatile_package_directories),
+        *([f"--volatile-package-directory={packages}"] if packages else []),
```

Comment:
The `packages` parameter name is not ideal. It made me do a double take here, since `packages` here is a directory, but `package` above are actual package names.